src/app/features/settings/go-pro/services/go-pro-wifi.service.ts

```diff
@@ -0,0 +1,65 @@
+import { Injectable } from '@angular/core';
+import { WifiPlugin } from '@capacitor-community/wifi';
+import { Plugins } from '@capacitor/core';
+import { Platform } from '@ionic/angular';
+import { GoProBluetoothService } from './go-pro-bluetooth.service';
+const Wifi: WifiPlugin = Plugins.Wifi as WifiPlugin;
+
+const { Storage } = Plugins;
```

Comment:
The App already has a `PreferenceManager` class which wraps the `Storage` plugin and makes values as subscriptables. Please use `PreferenceManager` directly instead of access `Storage` plugin directly here so that the App has a single uniform way to handle key-value storages.
You can check `WebCryptoApiSignatureProvider` for usage example for the `PreferenceManager`.

Reply:
https://github.com/numbersprotocol/capture-lite/pull/1205/commits/1688367f403514d557a618b10b7597981729bde0 using PreferenceManager for WiFi

https://github.com/numbersprotocol/capture-lite/pull/1205/commits/4e5d9a3a4cf4621f6e6308064b590160c33c69f5 using PreferenceManager for BLE

However running locally `npm run test.ci` giving this error:


after commit https://github.com/numbersprotocol/capture-lite/pull/1205/commits/1688367f403514d557a618b10b7597981729bde0
<details>
<summary>test logs</summary>

```
GoProWifiService should be created FAILED
	NullInjectorError: R3InjectorError(DynamicTestModule)[GoProWifiService -> PreferenceManager -> InjectionToken STORAGE_PLUGIN -> InjectionToken STORAGE_PLUGIN]: 
	  NullInjectorError: No provider for InjectionToken STORAGE_PLUGIN!
	error properties: Object({ ngTempTokenPath: null, ngTokenPath: [ 'GoProWifiService', 'PreferenceManager', 'InjectionToken STORAGE_PLUGIN', 'InjectionToken STORAGE_PLUGIN' ] })
	NullInjectorError: R3InjectorError(DynamicTestModule)[GoProWifiService -> PreferenceManager -> InjectionToken STORAGE_PLUGIN -> InjectionToken STORAGE_PLUGIN]: 
	  NullInjectorError: No provider for InjectionToken STORAGE_PLUGIN!
	    at NullInjector.get (http://localhost:9876/_karma_webpack_/vendor.js:74891:27)
	    at R3Injector.get (http://localhost:9876/_karma_webpack_/vendor.js:75058:33)
	    at R3Injector.get (http://localhost:9876/_karma_webpack_/vendor.js:75058:33)
	    at injectInjectorOnly (http://localhost:9876/_karma_webpack_/vendor.js:68541:33)
	    at ɵɵinject (http://localhost:9876/_karma_webpack_/vendor.js:68545:61)
	    at Object.PreferenceManager_Factory [as factory] (ng:///PreferenceManager/ɵfac.js:4:45)
	    at R3Injector.hydrate (http://localhost:9876/_karma_webpack_/vendor.js:75228:35)
	    at R3Injector.get (http://localhost:9876/_karma_webpack_/vendor.js:75047:33)
	    at injectInjectorOnly (http://localhost:9876/_karma_webpack_/vendor.js:68541:33)
	    at ɵɵinject (http://localhost:9876/_karma_webpack_/vendor.js:68545:61)
	Error: Expected undefined to be truthy.
	    at <Jasmine>
	    at UserContext.<anonymous> (http://localhost:9876/_karma_webpack_/main.js:1131:25)
	    at ZoneDelegate.invoke (http://localhost:9876/_karma_webpack_/polyfills.js:8285:26)
	    at ProxyZoneSpec.onInvoke (http://localhost:9876/_karma_webpack_/vendor.js:340077:39)
```
</details>


after commit https://github.com/numbersprotocol/capture-lite/pull/1205/commits/4e5d9a3a4cf4621f6e6308064b590160c33c69f5
<details>
<summary>test logs</summary>

```
[31mChrome Headless 98.0.4758.80 (Mac OS 10.15.7) GoProBluetoothService should be created FAILED
	NullInjectorError: R3InjectorError(DynamicTestModule)[GoProBluetoothService -> PreferenceManager -> InjectionToken STORAGE_PLUGIN -> InjectionToken STORAGE_PLUGIN]: 
	  NullInjectorError: No provider for InjectionToken STORAGE_PLUGIN!
	error properties: Object({ ngTempTokenPath: null, ngTokenPath: [ 'GoProBluetoothService', 'PreferenceManager', 'InjectionToken STORAGE_PLUGIN', 'InjectionToken STORAGE_PLUGIN' ] })
	NullInjectorError: R3InjectorError(DynamicTestModule)[GoProBluetoothService -> PreferenceManager -> InjectionToken STORAGE_PLUGIN -> InjectionToken STORAGE_PLUGIN]: 
	  NullInjectorError: No provider for InjectionToken STORAGE_PLUGIN!
	    at NullInjector.get (http://localhost:9876/_karma_webpack_/vendor.js:74891:27)
	    at R3Injector.get (http://localhost:9876/_karma_webpack_/vendor.js:75058:33)
	    at R3Injector.get (http://localhost:9876/_karma_webpack_/vendor.js:75058:33)
	    at injectInjectorOnly (http://localhost:9876/_karma_webpack_/vendor.js:68541:33)
	    at ɵɵinject (http://localhost:9876/_karma_webpack_/vendor.js:68545:61)
	    at Object.PreferenceManager_Factory [as factory] (ng:///PreferenceManager/ɵfac.js:4:45)
	    at R3Injector.hydrate (http://localhost:9876/_karma_webpack_/vendor.js:75228:35)
	    at R3Injector.get (http://localhost:9876/_karma_webpack_/vendor.js:75047:33)
	    at injectInjectorOnly (http://localhost:9876/_karma_webpack_/vendor.js:68541:33)
	    at ɵɵinject (http://localhost:9876/_karma_webpack_/vendor.js:68545:61)
	Error: Expected undefined to be truthy.
	    at <Jasmine>
	    at UserContext.<anonymous> (http://localhost:9876/_karma_webpack_/main.js:1077:25)
	    at ZoneDelegate.invoke (http://localhost:9876/_karma_webpack_/polyfills.js:8285:26)
	    at ProxyZoneSpec.onInvoke (http://localhost:9876/_karma_webpack_/vendor.js:340077:39)
```
</details>